packages/css/button.css

```diff
@@ -1,202 +1,171 @@
-    background-color: var(--dsc-btn-tertiary-hover-background);
+  &[data-ds-color="danger"][data-ds-variant="secondary"] {
+    --dsc-button-background--active: var(--ds-color-danger-surface-hover);
+    --dsc-button-background--hover: var(--ds-color-danger-surface-default);
+    --dsc-button-background: transparent;
+    --dsc-button-border-color: var(--ds-color-danger-border-strong);
+    --dsc-button-color--active: var(--ds-color-danger-text-default);
+    --dsc-button-color--hover: var(--ds-color-danger-text-default);
+    --dsc-button-color: var(--ds-color-danger-text-subtle);
   }
-}
 
-/* Primary button colors */
-.ds-btn--primary {
-  background-color: var(--dsc-btn-primary-background);
-}
+  &[data-ds-color="danger"][data-ds-variant="tertiary"] {
+    --dsc-button-background--active: var(--ds-color-danger-surface-hover);
+    --dsc-button-background--hover: var(--ds-color-danger-surface-default);
+    --dsc-button-background: transparent;
+    --dsc-button-color--active: var(--ds-color-danger-text-default);
+    --dsc-button-color--hover: var(--ds-color-danger-text-default);
+    --dsc-button-color: var(--ds-color-danger-text-subtle);
+  }
 
-.ds-btn--primary:not([aria-disabled='true'], :disabled):active {
-  background-color: var(--dsc-btn-primary-active-background);
-}
+  /**
+    * States
+    */
 
-/* Secondary button colors */
-.ds-btn--secondary {
-  color: var(--dsc-btn-secondary-color);
-  border-color: var(--dsc-btn-secondary-border-color);
-  background-color: transparent;
-}
+  @media (hover: hover) and (pointer: fine) { /* Only use hover for non-touch devices to prevent sticky-hovering */
+    &:not(:disabled, [aria-disabled="true"], [aria-busy="true"]):hover {
+      background: var(--dsc-button-background--hover);
+    }
+  }
 
-.ds-btn--secondary:not([aria-disabled='true'], :disabled):active {
-  color: var(--dsc-btn-secondary-active-color);
-  border-color: var(--dsc-btn-secondary-border-color);
-  background-color: var(--dsc-btn-secondary-active-background);
-}
+  &[aria-busy="true"]{
+    cursor: progress;
+  }
 
-/* Tertiary button colors */
-.ds-btn--tertiary {
-  color: var(--dsc-btn-tertiary-color);
-}
+  &[aria-busy="true"]::before {
+    background: currentcolor;
+    content: '';
+    height: 1.5em;
+    margin-inline: var(--dsc-button-icon-adjust);
+    mask: center/contain no-repeat url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 50 50'%3E%3Cstyle%3E@keyframes rotate%7Bto%7Brotate:360deg%7D%7D@keyframes stroke%7B50%25%7Bstroke-dasharray:100,200;stroke-dashoffset:-15;transform:rotate(0deg)%7Dto%7Bstroke-dasharray:1,200;stroke-dashoffset:-120;transform:rotate(15deg)%7D%7Dcircle%7Bcx:25;cy:25;r:20;fill:none;stroke:currentcolor;stroke-width:5%7Dcircle:last-child%7Banimation:stroke ease-in-out infinite 2s,rotate linear infinite 2s;transform-origin:center%7D%3C/style%3E%3Ccircle opacity='.15'/%3E%3Ccircle stroke-dasharray='1, 200'/%3E%3C/svg%3E");
```

Comment:
I don't think we should put a spinner here for cosumers. It is up to them to put the spinner they want here.
If they create their own variant, they might want a spinner with their own colors.
How does color work on this? If it is `currentColor`, it would easily break for a danger secondary button.
![image](https://github.com/user-attachments/assets/1ee68e76-4dd2-4ac7-ab5f-fb421aa85288)

If we make changes to our spinner, it would not be reflected here

Reply:
Dette ble gjennomgått med @Febakke, men dere må gjerne diskutere mer.
Jeg tenker det er en styrke at knappen er selvstendig - også uten React og uten å måtte legge inn komponenter, men her styrer dere